pipeline/authn/authenticator_oauth2_introspection.go

```diff
@@ -155,7 +181,14 @@ func (a *AuthenticatorOAuth2Introspection) Authenticate(r *http.Request, session
 		}
 		// set/override the content-type header
 		introspectReq.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+
+		// add tracing
+		closeSpan := a.traceRequest(r.Context(), introspectReq)
+
```

Comment:
Should probably be:

```
defer a.traceRequest(r.Context(), introspectReq)
```

Reply:
If we use `defer` here then we'll be including all of the later code inside the trace span, which may lead to an inflated duration of the span. The later call `tokenToCache` - is that just an in memory cache. If so then we could use `defer`, but making the call to `closeSpan` explicit means theres zero chance of later adding code that inflates the span duration.

Reply:
My bad, that makes total sense!